src/rez/utils/formatting.py

```diff
@@ -14,7 +14,7 @@
 import time
 
 
-PACKAGE_NAME_REGSTR = "[a-zA-Z_0-9](\.?[a-zA-Z0-9_]+)*"
+PACKAGE_NAME_REGSTR = "[a-zA-Z_0-9](\.?[a-zA-Z0-9_-]+)*"
```

Comment:
Again I'm not quite clear on what's going on here.

The problem with this change is that in rez, `-` _always_ separates the package name from its version. If a package name can contain hyphen, then which part is the name and which is the version becomes ambiguous.

Reply:
Hello Allan, yes this is true. This commit was from some time ago before I realized that rez uses dash to separate the package name from its version. After some tests it does not seem to be relevant to the SchemaError that occurs with some packages. I will revert this change and provide an example to the SchemaError.